tlatools/org.lamport.tlatools/src/tlc2/util/Context.java

```diff
@@ -236,6 +239,23 @@ public Context deepCopy() {
 		}
 		return new Context(this.name, this.value, this.next.deepCopy());
 	}
+
+	public Context replaceSuffix(Context oldSuffix, Context newSuffix) {
```

Comment:
Does this mutate the existing `Context` instance?  If not, should this rather be static?

Reply:
It constructs a new `Context`. Contexts are (thankfully!) immutable.

Perhaps `withDifferentSuffix` might be a better name? `with` seems to be the word Java developers usually use for methods that return a modified copy (as opposed to modifying in-place).

Reply:
Or perhaps a new constructor `Context(Context orig, Context oldSuffix, Context newSuffix)`?